### PR TITLE
pybibget: init at 0.1.0

### PR DESCRIPTION
**Pull request by Vasissualiyp**
## Description of changes

A python package to obtain .bib information about papers from doi, arxiv, etc.
[Homepage](https://github.com/wirhabenzeit/pybibget)

Added myself as a maintainer. Please, let me know if there are any issues.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/pybibget/default.nix
Comment (Aleksanaa) [inline pkgs/development/python-modules/pybibget/default.nix:19]: please add `pyproject = true` here

File: pkgs/development/python-modules/pybibget/default.nix
Comment (Aleksanaa) [inline pkgs/development/python-modules/pybibget/default.nix:17]: Please add `pyproject = true` here

Comment (Vasissualiyp) [reply pkgs/development/python-modules/pybibget/default.nix:17]: Properly formatted commit messages and added pyproject as requested

**Comment by eclairevoyant** [comment]
@ofborg build python311Packages.pybibget python312Packages.pybibget

**Comment by eclairevoyant** [comment]
wait, is this a CLI or a library? if it's solely the former, it should go in `pkgs/by-name` and use [`buildPythonApplication`](https://nixos.org/manual/nixpkgs/unstable/#buildpythonapplication-function)

File: pkgs/by-name/py/pybibget/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/py/pybibget/package.nix:37]: These comments can be cleaned up a bit, if we're disabling checks, just add a comment explaining why

File: pkgs/by-name/py/pybibget/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/py/pybibget/package.nix:43]:
```suggestion
  meta = {
    description = "Command line utility to automatically retrieve BibTeX citations from MathSciNet, arXiv, PubMed and doi.org";
    homepage = "https://github.com/wirhabenzeit/pybibget";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ vasissualiyp ];
```

Tracking issue: #292468
Any uses of `lib` are better "pushed down"

**Review by eclairevoyant — COMMENTED** [review]
Overall looks good, tested and ran the binaries, they seem to work.
Couple points remaining and I'll approve after.

**Comment by Vasissualiyp** [comment]
Thank you for your time!
Have force-pushed all the requested changes.

**Review by eclairevoyant — APPROVED** [review]

**Comment by eclairevoyant** [comment]
@ofborg eval

**Comment by Vasissualiyp** [comment]
It has been sitting here for 2 months, all checks pass, I was wondering when we could merge this?

**Comment by eclairevoyant** [comment]
Apologies, was waiting for CI to clear and it fell off the radar. Thanks for your contribution!

**Pull request merged**
